### PR TITLE
Add email ingestion API with Claude-powered parsing

### PR DESCRIPTION
**Pull request by qiheme**
## Summary
This PR adds a new email ingestion API endpoint that accepts travel confirmation emails, uses Claude to parse them into structured booking data, and stores the results in the inbox. It also adds real-time subscription support for inbox changes.

### New Ingestion API (`apps/web/api/inbox/ingest.ts`)
- Created `POST /api/inbox/ingest` endpoint that:
  - Validates incoming requests with secret-based authentication
  - Accepts email metadata (subject, from, body, received date)
  - Creates a placeholder inbox item with `parsing` status
  - Calls Claude Haiku to extract booking details (vendor, type, dates, cost, confirmation number)
  - Matches parsed bookings to user's active trips with confidence scoring
  - Updates inbox item with parsed data or marks as `needs_review` if confidence < 0.5 or parsing fails
  - Returns gracefully on errors (sets status to `needs_review` with error note)

### Database Functions (`apps/web/src/lib/db.ts`)
- Added `insertInboxItem()` to insert new inbox items from the API
- Added `subscribeInbox()` for real-time inbox updates via Supabase Postgres Changes
- Extracted `rowToInboxItem()` helper to map database rows to `InboxItem` objects
- Subscription handles INSERT, UPDATE, and DELETE events with proper field mapping (`from_address` → `from`)

### UI Integration (`apps/web/src/app/AppContext.tsx`)
- Integrated inbox subscription into `AppProvider`
- Subscribes to inbox changes when user authenticates
- Unsubscribes when user logs out
- Updates local inbox state on INSERT, UPDATE, and DELETE events

### Testing
- Added comprehensive test suite (`apps/web/api/inbox/ingest.test.ts`) with 395 lines covering:
  - Authentication and authorization
  - Request validation
  - Happy path with trip matching
  - Error handling (invalid JSON, API failures, parsing errors)
  - Edge cases (null confidence, non-text Claude responses, missing optional fields)
- Updated `apps/web/src/lib/db.test.ts` with tests for new `insertInboxItem()` and `subscribeInbox()` functions

### Configuration
- Added `@anthropic-ai/sdk` dependency
- Updated TypeScript config to include Node types for crypto module
- Updated Vite coverage config to include API routes

## Implementation Details
- Uses Claude Haiku 4.5 for cost-effective parsing
- Implements graceful degradation: parsing errors don't fail the request, just mark item for review
- Supports optional `source` field (defaults to "email")
- Handles null/undefined confidence values (defaults to 0)
- Provides detailed error notes when parsing fails

https://claude.ai/code/session_01DhKZY3rczUKnNeXdSg5v5t

### CODE REVIEW

File: apps/web/api/inbox/ingest.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+function isValidBody(b: unknown): b is IngestBody {
+  if (!b || typeof b !== 'object') return false;
+  const o = b as Record<string, unknown>;
+  return (
+    typeof o['userId'] === 'string' &&
+    !!o['raw'] &&
+    typeof (o['raw'] as Record<string, unknown>)['subject'] === 'string' &&
+    typeof (o['raw'] as Record<string, unknown>)['text'] === 'string'
```
[the remark this answers is not in the document]
Comment (qiheme) [reply apps/web/api/inbox/ingest.ts]: Fixed in edc41d3 — `isValidBody` now requires `raw.from` and `raw.receivedAt` as strings, so malformed payloads are rejected with 400 before any DB write. Added tests for each missing-field case plus the userId type guard. 👍

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/api/inbox/ingest.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      return Response.json({ id }, { status: 200 });
+    }
+
+    const confidence = parsed.suggested_confidence ?? 0;
+    const status = confidence >= 0.5 ? 'parsed' : 'needs_review';
+    await supabase.from('inbox_items').update({
+      status,
+      vendor: parsed.vendor,
+      parsed: {
+        type: parsed.type,
```
[the remark this answers is not in the document]
Comment (qiheme) [reply apps/web/api/inbox/ingest.ts]: Fixed in edc41d3 — the parsed-row update now destructures `error`; on failure, returns 500 with the placeholder id so upstream callers can retry. Added a test that exercises this branch. 👍

---
_Generated by [Claude Code](https://claude.ai/code)_
